executor/tests/test_launcher.py

```diff
@@ -0,0 +1,96 @@
+import os
+import glob
+
+from queue import Queue
+from threading import Event, Thread
+
+from cook.launcher import run_commands
+
+def readlines(path):
+    with open(path) as f:
+        return [line.rstrip('\n') for line in f]
+
+def remove_logs():
+    for path in glob.glob("stdout*"):
+        os.remove(path)
+    for path in glob.glob("stderr*"):
+        os.remove(path)
+
+def setup_module(module):
+    remove_logs()
+
+def teardown_module(module):
+    remove_logs()
+
+def test_run_commands_simple():
+    codes = run_commands([
+        {'name': 0, 'value': 'echo 1'},
+        {'name': 1, 'value': 'echo 2'}
+    ])
+
+    assert codes == [0, 0]
+
+def test_run_commands_with_guard():
+    """
+    A failing guard should prevent the execution of additional commands
+    """
+    codes = run_commands([
+        {'name': 0, 'value': '/bin/sh -c "exit 1"', 'guard': True},
+        {'name': 1, 'value': 'echo 1'}
+    ])
+
+    assert codes == [1]
+
+def test_run_commands_with_async():
```

Comment:
Should we add some more unit tests testing more complex scenarios?  Multiple async, multiple guards, guards combined with asyncs, etc?